### ISSUE TITLE
Create a custom modifier using a composable modifier factory is now recommended, however it is still failing "Avoid Modifier extension factory functions"

### ISSUE THREAD

**Issue by YingChen0605** [twitter/compose-rules#130]
**Describe the bug**

https://twitter.github.io/compose-rules/rules/#avoid-modifier-extension-factory-functions is no longer true now. Refer to the [official documentation](https://developer.android.com/develop/ui/compose/custom-modifiers#:~:text=Note%3A%20In%20previous%20versions%20of%20Compose%2C%20we%20recommended%20against%20this%20approach%20and%20suggested%20using%20composed%20%7B%7D%20instead%20via%20a%20lint%20rule.%20Now%20that%20composed%20%7B%7D%20is%20not%20recommended%2C%20the%20lint%20rule%20has%20been%20remove):
>  Note: In previous versions of Compose, we recommended against this approach and suggested using [composed {}](https://developer.android.com/reference/kotlin/androidx/compose/ui/Modifier#(androidx.compose.ui.Modifier).composed(kotlin.Function1,kotlin.Function1)) instead via a lint rule. Now that composed {} is not recommended, the lint rule has been removed.

**To Reproduce**
Adding a new Composable:
```
@Composable
private fun Modifier.modify(viewModel: xxViewModel) =  this then Modifier.viewable(onView = viewModel.onView)
        .margin(margin = viewModel.margin)
```

**Expected behavior**
It should not be flagged as a violation.

**Screenshots**
<img width="1417" alt="Screenshot 2024-08-05 at 12 21 31 PM" src="https://github.com/user-attachments/assets/e7e275e2-4e13-45be-b827-601b908683ad">

**Environment**
com.twitter.compose.rules.detekt of 0.0.26

**Comment by kenyee** [comment]
These have migrated to:
https://github.com/mrmans0n/compose-rules
https://slackhq.github.io/compose-lints/

**Comment by YingChen0605** [comment]
@kenyee Thank you for getting back so quickly and informing about the migration! 👍 
Assuming https://github.com/twitter/[compose-rules](https://github.com/twitter/compose-rules) is now deprecated.
Would you suggest choosing between https://github.com/mrmans0n/compose-rules(ktlint and detekt) and
https://slackhq.github.io/compose-lints/(lint) for use? Do their rules overlap to some extent?

**Comment by kenyee** [comment]
use the one that fits your usage best.  Slack's is done using Android lint.  The original was ktlint+detekt.
